BUG: Maven doesn't build a compat app for Java 8

Deploy a `compat` runtime fails if it's java8.

My` appengine-web.xml`:
```
<appengine-web-app xmlns="http://appengine.google.com/ns/1.0">
  <application>your-project-id</application>
  <vm>true</vm>
  <threadsafe>true</threadsafe>
</appengine-web-app>
```

```
mvn appengine:run
[INFO] --- maven-war-plugin:2.2:war (default-war) @ helloworld-compat ---
[INFO] Packaging webapp
[INFO] Assembling webapp [helloworld-compat] in [/Users/lesv/github/getting-started-java/helloworld-compat/target/helloworld-compat-1.0-SNAPSHOT]
[INFO] Processing war project
[INFO] Copying webapp resources [/Users/lesv/github/getting-started-java/helloworld-compat/src/main/webapp]
[INFO] Webapp assembled in [101 msecs]
[INFO] Building war: /Users/lesv/github/getting-started-java/helloworld-compat/target/helloworld-compat-1.0-SNAPSHOT.war
[INFO] WEB-INF/web.xml already added, skipping
[INFO]
[INFO] <<< appengine-maven-plugin:0.1.1-beta:deploy (default-cli) < package @ helloworld-compat <<<
[INFO]
[INFO] --- appengine-maven-plugin:0.1.1-beta:deploy (default-cli) @ helloworld-compat ---
[INFO] Staging the application to: /Users/lesv/github/getting-started-java/helloworld-compat/target/appengine-staging
[INFO] Detected App Engine standard environment application.
Aug 30, 2016 5:41:59 PM com.google.cloud.tools.appengine.cloudsdk.CloudSdk logCommand
INFO: submitting command: /Library/Java/JavaVirtualMachines/jdk1.8.0_91.jdk/Contents/Home/jre/bin/java -cp /Users/lesv/google-cloud-sdk/platform/google_appengine/google/appengine/tools/java/lib/appengine-tools-api.jar com.google.appengine.tools.admin.AppCfg --disable_update_check stage /Users/lesv/github/getting-started-java/helloworld-compat/target/helloworld-compat-1.0-SNAPSHOT /Users/lesv/github/getting-started-java/helloworld-compat/target/appengine-staging
[INFO] GCLOUD: Reading application configuration data...
[INFO] GCLOUD:
[INFO] GCLOUD:
[INFO] GCLOUD: Beginning interaction for module default...
[INFO] GCLOUD: Unable to stage app: Class file is Java 8 but max supported is Java 7: /Users/lesv/github/getting-started-java/helloworld-compat/target/helloworld-compat-1.0-SNAPSHOT/WEB-INF/classes/com/example/appengine/helloworld/HelloServlet.class
[INFO] GCLOUD: Please see the logs [/var/folders/xr/8psp9665483gvdbs6t2jl3s00044fd/T/appcfg5048155306970265780.log] for further information.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.934 s
[INFO] Finished at: 2016-08-30T17:42:00-07:00
[INFO] Final Memory: 19M/305M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.google.cloud.tools:appengine-maven-plugin:0.1.1-beta:deploy (default-cli) on project helloworld-compat: Execution default-cli of goal com.google.cloud.tools:appengine-maven-plugin:0.1.1-beta:deploy failed: Non zero exit: 1 -> [Help 1]
```

**Log**
```
java.lang.IllegalArgumentException: Class file is Java 8 but max supported is Java 7: /Users/lesv/github/getting-started-java/helloworld-compat/target/helloworld-compat-1.0-SNAPSHOT/WEB-INF/classes/com/example/appengine/helloworld/HelloServlet.class
        at com.google.appengine.tools.admin.Application.checkJavaVersion(Application.java:1378)
        at com.google.appengine.tools.admin.Application.checkJavaClassVersion(Application.java:1347)
        at com.google.appengine.tools.admin.Application.checkJavaVersion(Application.java:1329)
        at com.google.appengine.tools.admin.Application.copyOrLinkFile(Application.java:1256)
        at com.google.appengine.tools.admin.Application.copyOrLink(Application.java:1235)
        at com.google.appengine.tools.admin.Application.copyOrLink(Application.java:1224)
        at com.google.appengine.tools.admin.Application.copyOrLink(Application.java:1224)
        at com.google.appengine.tools.admin.Application.copyOrLink(Application.java:1224)
        at com.google.appengine.tools.admin.Application.copyOrLink(Application.java:1224)
        at com.google.appengine.tools.admin.Application.copyOrLink(Application.java:1224)
        at com.google.appengine.tools.admin.Application.copyOrLink(Application.java:1222)
        at com.google.appengine.tools.admin.Application.populateStagingDirectory(Application.java:756)
        at com.google.appengine.tools.admin.Application.createStagingDirectory(Application.java:741)
        at com.google.appengine.tools.admin.AppAdminImpl.stageApplication(AppAdminImpl.java:531)
        at com.google.appengine.tools.admin.AppAdminImpl.stageApplicationWithDefaultResourceLimits(AppAdminImpl.java:484)
        at com.google.appengine.tools.admin.AppCfg$StagingAction.execute(AppCfg.java:2473)
        at com.google.appengine.tools.admin.AppCfg.executeAction(AppCfg.java:357)
        at com.google.appengine.tools.admin.AppCfg.<init>(AppCfg.java:218)
        at com.google.appengine.tools.admin.AppCfg.<init>(AppCfg.java:119)
        at com.google.appengine.tools.admin.AppCfg.main(AppCfg.java:115)
com.google.appengine.tools.admin.AdminException: Unable to stage app: Class file is Java 8 but max supported is Java 7: /Users/lesv/github/getting-started-java/helloworld-compat/target/helloworld-compat-1.0-SNAPSHOT/WEB-INF/classes/com/example/appengine/helloworld/HelloServlet.class
        at com.google.appengine.tools.admin.AppAdminImpl.stageApplication(AppAdminImpl.java:535)
        at com.google.appengine.tools.admin.AppAdminImpl.stageApplicationWithDefaultResourceLimits(AppAdminImpl.java:484)
        at com.google.appengine.tools.admin.AppCfg$StagingAction.execute(AppCfg.java:2473)
        at com.google.appengine.tools.admin.AppCfg.executeAction(AppCfg.java:357)
        at com.google.appengine.tools.admin.AppCfg.<init>(AppCfg.java:218)
        at com.google.appengine.tools.admin.AppCfg.<init>(AppCfg.java:119)
        at com.google.appengine.tools.admin.AppCfg.main(AppCfg.java:115)
Caused by: java.lang.IllegalArgumentException: Class file is Java 8 but max supported is Java 7: /Users/lesv/github/getting-started-java/helloworld-compat/target/helloworld-compat-1.0-SNAPSHOT/WEB-INF/classes/com/example/appengine/helloworld/HelloServlet.class
        at com.google.appengine.tools.admin.Application.checkJavaVersion(Application.java:1378)
        at com.google.appengine.tools.admin.Application.checkJavaClassVersion(Application.java:1347)
        at com.google.appengine.tools.admin.Application.checkJavaVersion(Application.java:1329)
        at com.google.appengine.tools.admin.Application.copyOrLinkFile(Application.java:1256)
        at com.google.appengine.tools.admin.Application.copyOrLink(Application.java:1235)
        at com.google.appengine.tools.admin.Application.copyOrLink(Application.java:1224)
        at com.google.appengine.tools.admin.Application.copyOrLink(Application.java:1224)
        at com.google.appengine.tools.admin.Application.copyOrLink(Application.java:1224)
        at com.google.appengine.tools.admin.Application.copyOrLink(Application.java:1224)
        at com.google.appengine.tools.admin.Application.copyOrLink(Application.java:1224)
        at com.google.appengine.tools.admin.Application.copyOrLink(Application.java:1222)
        at com.google.appengine.tools.admin.Application.populateStagingDirectory(Application.java:756)
        at com.google.appengine.tools.admin.Application.createStagingDirectory(Application.java:741)
        at com.google.appengine.tools.admin.AppAdminImpl.stageApplication(AppAdminImpl.java:531)
        ... 6 more
```

```
mvn appengine:run
[INFO] --- maven-war-plugin:2.2:war (default-war) @ helloworld-compat ---
[INFO] Packaging webapp
[INFO] Assembling webapp [helloworld-compat] in [/Users/lesv/github/getting-started-java/helloworld-compat/target/helloworld-compat-1.0-SNAPSHOT]
[INFO] Processing war project
[INFO] Copying webapp resources [/Users/lesv/github/getting-started-java/helloworld-compat/src/main/webapp]
[INFO] Webapp assembled in [16 msecs]
[INFO] Building war: /Users/lesv/github/getting-started-java/helloworld-compat/target/helloworld-compat-1.0-SNAPSHOT.war
[INFO] WEB-INF/web.xml already added, skipping
[INFO]
[INFO] <<< appengine-maven-plugin:0.1.1-beta:run (default-cli) < package @ helloworld-compat <<<
[INFO]
[INFO] --- appengine-maven-plugin:0.1.1-beta:run (default-cli) @ helloworld-compat ---
Aug 30, 2016 5:39:19 PM com.google.cloud.tools.appengine.cloudsdk.CloudSdk logCommand
INFO: submitting command: /Users/lesv/google-cloud-sdk/bin/dev_appserver.py /Users/lesv/github/getting-started-java/helloworld-compat/target/helloworld-compat-1.0-SNAPSHOT --jvm_flag=-Dappengine.user.timezone=UTC
[INFO] GCLOUD: INFO     2016-08-31 00:39:20,432 application_configuration.py:431] No version specified. Generated version id: 20160831t003920
[INFO] GCLOUD: INFO     2016-08-31 00:39:20,432 devappserver2.py:769] Skipping SDK update check.
[INFO] GCLOUD: INFO     2016-08-31 00:39:20,479 api_server.py:205] Starting API server at: http://localhost:62265
[INFO] GCLOUD: INFO     2016-08-31 00:39:20,484 dispatcher.py:197] Starting module "default" running at: http://localhost:8080
[INFO] GCLOUD: INFO     2016-08-31 00:39:20,488 admin_server.py:116] Starting admin server at: http://localhost:8000
[INFO] GCLOUD: 2016-08-31 00:39:20.799:INFO::main: Logging initialized @292ms
[INFO] GCLOUD: 2016-08-31 00:39:20.983:INFO:oejs.Server:main: jetty-9.2.10.v20150310
[INFO] GCLOUD: 2016-08-31 00:39:20.996:WARN:oejsh.RequestLogHandler:main: !RequestLog
[INFO] GCLOUD: 2016-08-31 00:39:20.997:INFO:oejdp.ScanningAppProvider:main: Deployment monitor [file:/Users/lesv/google-cloud-sdk/platform/google_appengine/google/appengine/tools/java/lib/jetty-base-sdk/contexts/] at interval 1
```

--- a/src/main/java/com/google/cloud/tools/maven/CloudSdkMojo.java
+++ b/src/main/java/com/google/cloud/tools/maven/CloudSdkMojo.java
@@ -71,14 +71,21 @@ public abstract class CloudSdkMojo extends AbstractMojo {
     // maven-plugin-compiler default is 1.5
     String javaVersion = "1.5";
     if (mavenProject != null) {
-      Plugin compilerPlugin = mavenProject
-          .getPlugin("org.apache.maven.plugins:maven-compiler-plugin");
-      if (compilerPlugin != null) {
-        Xpp3Dom config = (Xpp3Dom) compilerPlugin.getConfiguration();
-        if (config != null) {
-          Xpp3Dom domVersion = config.getChild("target");
-          if (domVersion != null) {
-            javaVersion = domVersion.getValue();
+      // check the maven.compiler.target property first
+      String mavenCompilerTargetProperty = mavenProject.getProperties()
+          .getProperty("maven.compiler.target");
+      if (mavenCompilerTargetProperty != null) {
+        javaVersion = mavenCompilerTargetProperty;
+      } else {
+        Plugin compilerPlugin = mavenProject
+            .getPlugin("org.apache.maven.plugins:maven-compiler-plugin");
+        if (compilerPlugin != null) {
+          Xpp3Dom config = (Xpp3Dom) compilerPlugin.getConfiguration();
+          if (config != null) {
+            Xpp3Dom domVersion = config.getChild("target");
+            if (domVersion != null) {
+              javaVersion = domVersion.getValue();
+            }
           }
         }
       }

